metafacture-csv/src/main/java/org/metafacture/csv/CsvEncoder.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright 2018-2023 Deutsche Nationalbibliothek et al
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.metafacture.csv;
+
+import org.metafacture.framework.FluxCommand;
+import org.metafacture.framework.MetafactureException;
+import org.metafacture.framework.ObjectReceiver;
+import org.metafacture.framework.StreamReceiver;
+import org.metafacture.framework.annotations.Description;
+import org.metafacture.framework.annotations.In;
+import org.metafacture.framework.annotations.Out;
+import org.metafacture.framework.helpers.DefaultStreamPipe;
+
+import com.opencsv.CSVWriter;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A csv encoder that converts a record into a csv line (Default separator: {@value #DEFAULT_SEP}).
+ *
+ * <p>
+ * Each record represents a row. Each literal value represents a column value.
+ * </P>
+ *
+ * @author eberhardtj (j.eberhardt@dnb.de)
+ */
+@Description("Encodes each value in a record as a csv row.")
+@In(StreamReceiver.class)
+@Out(String.class)
+@FluxCommand("encode-csv")
+public class CsvEncoder extends DefaultStreamPipe<ObjectReceiver<String>> {
+    public static final char DEFAULT_SEP = CSVWriter.DEFAULT_SEPARATOR;
+    private CSVWriter csvWriter;
+    private StringWriter writer;
+    private List<String> rowItems = new ArrayList<>();
+    private boolean isFirstRecord = true;
+    private List<String> header = new ArrayList<>();
+    private char separator = DEFAULT_SEP;
+    private boolean noQuotes;
+    private boolean includeHeader;
+    private boolean includeRecordId;
+
+    /**
+     * Creates an instance of {@link CsvEncoder} with a given separator.
+     *
+     * @param separator to separate columns
+     */
+    public CsvEncoder(final String separator) {
+        this.separator = separator.charAt(0);
+    }
+
+    /**
+     * Creates an instance of {@link CsvEncoder} with a given separator.
+     *
+     * @param separator to separate columns
+     */
+    public CsvEncoder(final char separator) {
+        this.separator = separator;
+    }
```

Comment:
Is this duplicate code?

Reply:
Not needed, since the java API automatically shows what type of parameter is needed - so there is implicitly two kinds of documenations: one with demanding `String`, one with `char`.